packages/engine-server/src/workspace/utils.ts

```diff
@@ -102,6 +102,28 @@ export class WorkspaceUtils {
     return dendronWorkspaceFolders.filter(isNotUndefined);
   }
 
+  /**
+   * Check if a file is a dendron note (vs a regular file or something else entirely)
+   */
+  static isDendronNote({
+    wsRoot,
+    vaults,
+    fpath,
+  }: { fpath: string } & WorkspaceOpts) {
+    // check if we have markdown file
+    if (!fpath.endsWith(".md")) {
+      return false;
+    }
+    // if markdown file, check if it is in a dendron vault
+    if (!WorkspaceUtils.isPathInWorkspace({ wsRoot, vaults, fpath })) {
+      return false;
+    }
+
+    // if markdown file, does it have frontmatter?
+    // this is a rough heuristic
+    return fs.readFileSync(fpath, { encoding: "utf8" }).startsWith("---");
```

Comment:
should we prefer async versions for perf reasons?

Reply:
changed it to `async`. something I do worry about is that its not immediately clear when functions are sync/async and we already have places in the code where we will either `await` a non-async function or fail to `await` an `async function`.

we should look into either:
- adding a convention (eg. isDendronNoteAsync) to be explicit about the return
- add stricter linting rules about asyncs

Reply:
I'm for an async convention, but additionally, how about explicitly stating its return value? `isDendronNote(...): Promise<string> {}`